### PR TITLE
feat: emit events for order trading volume

### PR DESCRIPTION
**Pull request by kogisin**
## Description

closes: #309 

## Tasks

- [x] Add events for order trading volume
- [x] Update spec docs

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Appropriate labels applied
- [ ] Targeted PR against correct branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://go.dev/blog/godoc).
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/liquidity/spec/07_events.md [hunk 1]
```diff
@@ -147,12 +147,24 @@ The `liquidity` module emits the following events:
...
-| order_result | request_id           | {reqId}              |
-| order_result | orderer              | {orderer}            |
-| order_result | pair_id              | {pairId}             |
-| order_result | order_direction      | {direction}          |
-| order_result | remaining_offer_coin | {remainingOfferCoin} |
-| order_result | received_coin        | {receivedCoin}       |
-| order_result | status               | {status}             |
+| Type              | Attribute Key        | Attribute Value      |
+|-------------------|----------------------|----------------------|
+| order_result      | request_id           | {reqId}              |
```
Comment (hallazzang) [inline x/liquidity/spec/07_events.md:152]:
```suggestion
| order_result      | request_id           | {orderId}            |
```

File: x/liquidity/types/events.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ const (
 	EventTypeDepositResult    = "deposit_result"
 	EventTypeWithdrawalResult = "withdrawal_result"
 	EventTypeOrderResult      = "order_result"
+	EventTypeUserOrderFilled  = "user_order_filled"
+	EventTypePoolOrderFilled  = "pool_order_filled"
```
Comment (hallazzang) [inline x/liquidity/types/events.go:17]:
```suggestion
	EventTypeUserOrderMatched = "user_order_matched"
	EventTypePoolOrderMatched = "pool_order_matched"
```

File: x/liquidity/types/events.go [hunk 3]
```diff
@@ -44,4 +46,6 @@ const (
 	AttributeKeyPairIds            = "pair_ids"
 	AttributeKeyCanceledOrderIds   = "canceled_order_ids"
 	AttributeKeyStatus             = "status"
+	AttributeKeyFilledAmount       = "filled_amount"
```
Comment (hallazzang) [inline x/liquidity/types/events.go:49]:
```suggestion
	AttributeKeyMatchedAmount      = "matched_amount"
```

**Review by hallazzang — CHANGES_REQUESTED** [review]

File: x/liquidity/keeper/swap.go [hunk 4]
```diff
@@ -415,22 +414,54 @@ func (k Keeper) ApplyMatchResult(ctx sdk.Context, pair types.Pair, orders []amm.
...
+			paidCoin := order.OfferCoin.Sub(order.RemainingOfferCoin)
+			receivedCoin := order.ReceivedDemandCoin
+
+			bulkOp.QueueSendCoins(pair.GetEscrowAddress(), order.ReserveAddress, sdk.NewCoins(receivedCoin))
+
+			ctx.EventManager().EmitEvents(sdk.Events{
+				sdk.NewEvent(
+					types.EventTypePoolOrderFilled,
+					sdk.NewAttribute(types.AttributeKeyOrderDirection, order.Direction.String()),
+					sdk.NewAttribute(types.AttributeKeyOrderer, order.ReserveAddress.String()),
```
[the remark this answers is not in the document]
Comment (kogisin) [reply x/liquidity/keeper/swap.go:458]: Let me make that change.

File: x/liquidity/keeper/swap.go [hunk 5]
```diff
@@ -415,22 +414,54 @@ func (k Keeper) ApplyMatchResult(ctx sdk.Context, pair types.Pair, orders []amm.
...
 				o.SetStatus(types.OrderStatusPartiallyMatched)
 				k.SetOrder(ctx, o)
-				// TODO: emit an event?
 			}
 			bulkOp.QueueSendCoins(pair.GetEscrowAddress(), order.Orderer, sdk.NewCoins(order.ReceivedDemandCoin))
+
+			ctx.EventManager().EmitEvents(sdk.Events{
+				sdk.NewEvent(
+					types.EventTypeUserOrderFilled,
+					sdk.NewAttribute(types.AttributeKeyOrderDirection, order.Direction.String()),
```
[the remark this answers is not in the document]
Comment (kogisin) [reply x/liquidity/keeper/swap.go:439]: Is this comment applicable to only user order?

[the remark this answers is not in the document]
Comment (hallazzang) [reply x/liquidity/keeper/swap.go:440]: Nope, we also need it for pools.

**Review by dongsam — APPROVED** [review]

**Pull request merged**